tests/subsys/compress/decompression/lzma/src/main.c

```diff
@@ -0,0 +1,559 @@
+	0xc0, 0xc4, 0xac, 0xc7, 0xac, 0x69, 0x37, 0x4b,
+	0x60, 0xb4, 0x87, 0xe9, 0x3d, 0x65, 0xcf, 0xa2,
+	0x4b, 0x2b, 0xef, 0xd0, 0xb9, 0xbf, 0xf9, 0xc9,
+	0x2f, 0x61, 0x52, 0x17, 0xca, 0x55, 0x03, 0x77
+};
+
+#if defined(CONFIG_NRF_COMPRESS_MEMORY_TYPE_MALLOC) && !defined(CONFIG_SOC_POSIX)
+static uint8_t *large_malloc_object = NULL;
+#endif
+
+/* Input valid lzma1 compressed data */
+const uint8_t dummy_data_lzma1_input[] = {
+#include "dummy_data_input_lzma1.inc"
+};
+
+ZTEST(nrf_compress_decompression, test_valid_implementation)
+{
+	struct nrf_compress_implementation *implementation = NULL;
+
+	implementation = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_LZMA);
+
+	zassert_not_equal(implementation, NULL, "Expected implementation to not be null\n");
+}
+
+ZTEST(nrf_compress_decompression, test_valid_implementation_elements)
+{
+	struct nrf_compress_implementation *implementation = NULL;
+
+	implementation = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_LZMA);
+
+	zassert_equal(implementation->id, NRF_COMPRESS_TYPE_LZMA,
+		      "Expected id element to have correct value\n");
+	zassert_not_equal(implementation->init, NULL, "Expected init element to not be null\n");
+	zassert_not_equal(implementation->deinit, NULL,
+			  "Expected deinit element to not be null\n");
+	zassert_not_equal(implementation->reset, NULL, "Expected reset element to not be null\n");
+	zassert_equal(implementation->compress, NULL, "Expected compress element to be null\n");
+	zassert_not_equal(implementation->decompress_bytes_needed, NULL,
+			  "Expected decompress_bytes_needed element to not be null\n");
+	zassert_not_equal(implementation->decompress, NULL,
+			  "Expected decompress to not be null\n");
+}
+
+ZTEST(nrf_compress_decompression, test_invalid_implementation)
+{
+	struct nrf_compress_implementation *implementation = NULL;
+
+	implementation = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_COUNT);
+
+	zassert_equal(implementation, NULL, "Expected implementation to be null\n");
+
+	implementation = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_COUNT + 300);
+
+	zassert_equal(implementation, NULL, "Expected implementation to be null\n");
+
+	implementation = nrf_compress_implementation_find(((uint16_t)-1));
+
+	zassert_equal(implementation, NULL, "Expected implementation to be null\n");
+}
+
+ZTEST(nrf_compress_decompression, test_valid_data_decompression)
+{
+	int rc;
+	uint32_t pos;
+	uint32_t offset;
+	uint8_t *output;
+	uint32_t output_size;
+	uint32_t total_output_size = 0;
+	uint8_t output_sha[SHA256_SIZE] = { 0 };
+	struct nrf_compress_implementation *implementation;
+	mbedtls_sha256_context ctx;
+
+	mbedtls_sha256_init(&ctx);
+        rc = mbedtls_sha256_starts(&ctx, false);
+	zassert_ok(rc, "Expected mbedtls sha256 start to be successful\n");
+
+	implementation = nrf_compress_implementation_find(NRF_COMPRESS_TYPE_LZMA);
+
+	pos = 0;
+
+	rc = implementation->init();
+	zassert_ok(rc, "Expected init to be successful\n");
```

Comment:
> Also it would be worth printing the rc in case when assert is triggered?

Is this a blocker? This is not generally done anywhere, most tests in zephyr do not do that and most tests in ncs do not do that (out of 476 instances in ncs, 4 print a variable)

Reply:
Dropped the \n, I remember it being required when writing mcumgr tests for things to appear on other lines, maybe that was a bug